docs/trex_tableau_viz_ref.md

```diff
@@ -0,0 +1,215 @@
+
+
+### `description`
+
+(Optional)    A description about the visualization. Currently, the description is not used anywhere and is not visible when the image is generated. However, it might be helpful to use this description as you comment your code.
+
+### `title`
+
+(Optional)    Creates a title for the image.
+
+```json
+title: "My Custom Viz", 
+
+```
+
+### `data`
+
+(Required) The data to use to create the image. The `data` consists of an array of objects, specified with the `"values"` key. Data is listed in row level order, where each object represents one item.
+
+```json
+data: {
+  values: [
+   { Category: 'A', Sales: 28 },
+   { Category: 'B', Sales: 55 },
+   { Category: 'C', Sales: 43 },
+   { Category: 'D', Sales: 91 },
+   { Category: 'E', Sales: 81 },
+   { Category: 'F', Sales: 53 },
+   { Category: 'G', Sales: 19 },
+   { Category: 'H', Sales: 87 },
+   { Category: 'I', Sales: 52 }
+  ]
+},
+
+```
+
+### `mark`
+
+(Optional) Supported mark types include: `bar`, `line`, `area`, `square`, `circle`, `text`. If no mark type is specified, the generated image will use the Tableau default for that data.
+
+
+### `markcolor`
+
+Specifies a mark color for all marks, for example, hex values. This is equivalent to selecting an arbitrary color in the Color Shelf of the Marks card in Tableau.
+
+```json
+ markcolor: "#FFA500",
+
+```
+
+### `size`
+
+(Optional) Specifies the size of the image in pixels. The size is described with width and height dimensions.
+
+| Key |  Value |
+|:--- |:--- |
+|`width` | Width of the SVG image. |
+|`height`| Height of the SVG image.|
+|`showlegend` | Boolean (`true`, `false`). Specifies whether to show or hide the Size legend. |
+
+If you specify the size, you must specify both the width and height. If you don't specify a size, the image uses the default size (600 x 400).
+
+```json
+"size": {"width": 800, "height": 600},
+
+```
+
+### `encoding`
+
+Corresponds to the Marks card in Tableau and describes how the marks (measures and dimensions) are encoded. In Tableau, when you drag fields onto the properties on the Marks card it encodes the data in the view. The `encoding` key lets you set the properties of the marks for the image. <br/> The `columns`, `rows` keys allow you to define the layout of your view, just like you do in Tableau today. You can also map fields to the `color`, `size`, and `text` properties. 
+Within these properties, you must specify the field to encode and its type (`continuous` or `discrete`). Some keys allow you to set additional values as shown in the following table.
+
+---
+
+#### `columns` and `rows`
+
+| Key |  Value |
+|:--- |:--- |
+|`field` | The name of the field to encode. |
+|`type`| The type of field, either `discrete` (blue "pill") or `continuous` (green "pill"). You can also specify the type using the enumerations: `tableau.VizImageEncodingType.Discrete` or `tableau.VizImageEncodingType.Continuous`.|
```

Comment:
Please have them use tableau.VizImageEncodingType rather than 'discrete' or 'continuous'. That will alleviate their frustration when they use 'discret' or 'continous'. All the examples should use the types also.

Reply:
I can update all these to use the enumerations. Just waiting to confirm that's what we want.

Reply:
I guess another plus of using the enums is that Visual Studio can give you auto completion.